packages/safety-suite-api/src/api/models/types.ts

```diff
@@ -944,3 +944,197 @@ export type LostRestrictedInput = {
+   */
+  onReport?: boolean | {key: string; values: unknown[]};
+  onColumn?: boolean;
+  /**
+   * Indirect modules which use this field.
+   */
+  indirectModules?: Alias[];
+}
+
+export type Required =
+  | boolean
+  | {
+      path: string[];
+      cases: {
+        pattern: string;
+        result: boolean;
+      }[];
+      default?: boolean;
+    };
+
+export type DateType = 'date' | 'time' | 'dateTime';
+
+export interface Field<Type extends FieldType = FieldType>
+  extends BaseField<Type> {
+  options: BaseFieldOptions<FieldTypes[Type]> & {
+    /**
+     * If the field is confidential.
+     */
+    protected: boolean;
+    /**
+     * If the field is required, can be a boolean or an object to be checked against config.
+     */
+    required: Required;
+    sideEffect?: unknown;
+  } & (Type extends 'list'
+      ? {
+          itemType: {
+            name: string;
+            type: FieldType;
+          };
+        }
+      : Type extends 'number'
+      ? {
+          /**
+           * Type of number. E.g. `decimal`
+           */
+          numberType?: string;
+          /**
+           * Minimum allowed value.
+           */
+          min?: number;
+        }
+      : Type extends 'dateTime'
+      ? {
+          errorOnHoliday?: boolean;
+          errorOnWeekend?: boolean;
+          onlyFuture?: boolean;
+          /**
+           * Format string. E.g. `MM/DD/YYYY`
+           */
+          format?: string;
+          type?: DateType;
+        }
+      : Type extends 'enum'
+      ? {
+          /**
+           * ID of the enum associated with this field.
+           */
+          enumSetId: number;
+          /**
+           * If the field can accept multiple values.
+           */
+          multi?: boolean;
+        }
+      : // eslint-disable-next-line @typescript-eslint/ban-types
+        {});
+}
+
+export interface Computation<ResultType extends FieldType = FieldType>
+  extends BaseField<'computed'> {
+  options: BaseFieldOptions<FieldTypes[ResultType]> & {
+    operation: string;
+    /**
+     * The data type of the computation result.
+     */
+    resultType?: ResultType;
+    ignoreBlank?: boolean;
+    separator?: string;
+    sortWith?: string;
+    status?: string;
+    /**
+     * When the computation will be run. E.g. `read`
+     */
+    computeOn?: string;
+    condition?: string;
+    ifTrue?: unknown;
+    ifFalse?: unknown;
+    arguments?: unknown[];
+    conditionArguments?: unknown[];
+  } & (ResultType extends 'number'
+      ? {
+          decimalPlaces?: number;
+        }
+      : ResultType extends 'dateTime'
+      ? {
+          dateType?: DateType;
+        }
+      : // eslint-disable-next-line @typescript-eslint/ban-types
+        {});
+}
+
+export interface Relation<Type extends RelationType = RelationType>
+  extends BaseField<Type> {
+  options: {
+    /**
+     * Array of template IDs the relation is for.
+     */
+    templatesId: Type extends 'singleModel' ? [number] : number[];
```

Comment:
This isn't strictly tied to single model / multi model, as that is the number of models (eg, 1 employee has many events).  But instead its around how many types of data that relation can have.  A situation where a single model can have multiple templates is our current CDL relation.  1 employee has 1 current CDL, but it can be an A-CDL, B-CDL, A Permit, B Permit or Non-CDL.

Reply:
However this must always be non-zero.  If the list is 0 size, then the json schema failed to run.

Reply:
Ahh that makes a lot of sense.